storage/remote/queue_manager.go

```diff
@@ -82,20 +91,42 @@ func newQueueManagerMetrics(r prometheus.Registerer, rn, e string) *queueManager
 		Help:        "Total number of samples successfully sent to remote storage.",
 		ConstLabels: constLabels,
 	})
+
+	m.succeededMetadataTotal = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace:   namespace,
+		Subsystem:   subsystem,
+		Name:        "succeeded_metadata_total",
+		Help:        "Total number of metadata entries successfully sent to remote storage.",
+		ConstLabels: constLabels,
+	})
 	m.failedSamplesTotal = prometheus.NewCounter(prometheus.CounterOpts{
 		Namespace:   namespace,
 		Subsystem:   subsystem,
 		Name:        "failed_samples_total",
 		Help:        "Total number of samples which failed on send to remote storage, non-recoverable errors.",
 		ConstLabels: constLabels,
 	})
+	m.failedMetadataTotal = prometheus.NewCounter(prometheus.CounterOpts{
+		Namespace:   namespace,
+		Subsystem:   subsystem,
+		Name:        "failed_metadata_total",
```

Comment:
Let's not compound existing instrumentation issues. Failed and total, not failed and success. It's more robust and easier to work with.

Reply:
Not sure I follow - taking a step back, what is the existing instrumentation issue?

Reply:
That it is exposing failed and success.

Reply:
Should I get rid of the new metrics and make them a label within `succeeded_samples_total` and `failed_samples_total`? Then, change the "type" label to something like `data_type`. Not using the retry metric. cc: @csmarchbanks

Reply:
No, get rid of succeeded and add a total one and then use labels consistently everywhere. We may aswell get it right while we're breaking things.

Reply:
The `succeeded_samples_total` metric has existed for at least 3 years. While we've added labels or changed label names since then, I'm not sure if we might want to think about deprecation policies for metrics that have existed for so long.

At the very least @gotjosh can you also update the mixin remote write dashboard in `documentation/` to reflect the metric name changes and new labels please.

Reply:
I would prefer not to include metadata in a metric called something like `*_samples_total`. I think those should be two separate metrics as it doesn't make much sense to aggregate them together (happy to have someone enlighten me if there is a good use case). 

Changing to the best practice of failure and total rather than success + failure is a good change. If the old metrics change as part of this PR we need to make sure it is listed in the change log at release time.

Reply:
We're planning to add at least one more data type to remote write, exemplars. If we continue with `samples_total`, `metadata_total, etc. we'll have to add a set of new metrics every time we add a new data type to the WAL/remote write. There may only be a few more types, but it's still unnecessary duplication and another spot to introduce human error.

Reply:
Are exemplars a separate data type, or would they just be a part of samples since they are defined per sample?

I like the rule of thumb from our [naming docs](https://prometheus.io/docs/practices/naming/#metric-names): "either the `sum()` or the `avg()` over all dimensions of a given metric should be meaningful." I don't see how the result of either of those operations are meaningful across metadata and samples, but I am happy to see an example. Maybe it would help me if you suggest a name for a metric that would span metatdata and samples?

Reply:
> I'm not sure if we might want to think about deprecation policies for metrics that have existed for so long.

There are none. We break metrics fairly regularly.

> I don't see how the result of either of those operations are meaningful across metadata and samples

The sum seems meaningful to me at least: Total RPCs sent by remote write.

My main objection here is inconsistency. In some places it's metric names and others it's labels - that doesn't make sense. I think a label is the right approach here.


I note that this PR as a whole is likely about to be superseded by #7771.

Reply:
> The sum seems meaningful to me at least: Total RPCs sent by remote write.

These metrics are not about RPCs though, they are about the total amount of samples/metadata entries that have been sent. If they were RPC level (such as bytes sent) I agree with you that it should be labels.

Reply:
Total entries sent via remote write sounds meaningful to me too.

Reply:
> I note that this PR as a whole is likely about to be superseded by #7771.

Awesome! I can see that #7771 is coming along nicely, should we close this in favour of focusing our efforts there?

Reply:
I'd suggest pausing for now rather than closing, just in case #7771 doesn't work out.

Reply:
Sounds good, if there's anything I might need to bring from here to there just let me know.

Reply:
@csmarchbanks Sounds like the we either want something like `remote_write_samples_sent_total` and `remote_write_metadata_sent_total`, or `remote_write_records_sent_total{type="sample|metadata"}`, rather than `samples_total` with labels for `sample|metadata`.

Reply:
Yep, either one is fine by me, and then make sure the bytes metric is consistent. Though as Brian mentioned in https://github.com/prometheus/prometheus/pull/6815#discussion_r465714167, `type` is typically avoided for the label name.

Reply:
I think we can mark this as resolved now.

Reply:
I am fine with the current setup.